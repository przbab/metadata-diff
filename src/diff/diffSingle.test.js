'use strict';

const { diffSingle, parseData, transformData } = require('./diffSingle');

jest.mock('../client', () => ({
    fetchPathname: async (pathname) => {
        const mockData = (type) => ({
            html: pathname,
            redirects: [
                {
                    status: 302,
                    url: `http://example.com/${type}`,
                    target: 'http://example.com',
                },
            ],
        });
        return {
            currentServerData: mockData('currentServer'),
            currentClientData: mockData('currentClient'),
            candidateServerData: mockData('candidateServer'),
            candidateClientData: mockData('candidateClient'),
        };
    },
}));

jest.mock('../parser', () => ({
    parse: async (...params) => params,
}));

describe('diff', () => {
    describe('diffSingle', () => {
        test(`should diff single pathname`, async () => {
            const diff = await diffSingle('/', {}, {});

            expect(diff).toMatchSnapshot();
        });
    });
    describe('parseData', () => {
        test(`should parse data in correct order`, async () => {
            const parsedData = await parseData(
                {
                    currentServerData: { html: 'currentServer' },
                    currentClientData: { html: 'currentClient' },
                    candidateServerData: { html: 'candidateServer' },
                    candidateClientData: { html: 'candidateClient' },
                },
                {}
            );

            expect(parsedData).toMatchSnapshot();
        });
    });
    describe('transformData', () => {
        test(`should transform redirects`, () => {
            const parsedData = {
                metadata: {},
                microdata: {},
                jsonLd: {},
            };
            const rawData = {
                html: '',
                redirects: [{ status: 301, url: 'http://example.com', target: 'http://example.com/test' }],
            };

            const transformedData = transformData(parsedData, rawData, {
                replaceBaseUrls: false,
            });

            expect(transformedData).toMatchSnapshot();
        });
    });
});
