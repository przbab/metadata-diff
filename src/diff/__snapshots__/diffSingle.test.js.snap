// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`diff diffSingle should diff single pathname 1`] = `
Object {
  "candidate": Object {
    "client": Object {
      "0": "/",
      "redirects": Array [
        Object {
          "status": 302,
          "target": "http://example.com",
          "url": "http://example.com/candidateClient",
        },
      ],
    },
    "server": Object {
      "0": "/",
      "redirects": Array [
        Object {
          "status": 302,
          "target": "http://example.com",
          "url": "http://example.com/candidateServer",
        },
      ],
    },
  },
  "client": Object {
    "candidate": Object {
      "0": "/",
      "redirects": Array [
        Object {
          "status": 302,
          "target": "http://example.com",
          "url": "http://example.com/candidateClient",
        },
      ],
    },
    "current": Object {
      "0": "/",
      "redirects": Array [
        Object {
          "status": 302,
          "target": "http://example.com",
          "url": "http://example.com/currentClient",
        },
      ],
    },
  },
  "pathname": "/",
  "server": Object {
    "candidate": Object {
      "0": "/",
      "redirects": Array [
        Object {
          "status": 302,
          "target": "http://example.com",
          "url": "http://example.com/candidateServer",
        },
      ],
    },
    "current": Object {
      "0": "/",
      "redirects": Array [
        Object {
          "status": 302,
          "target": "http://example.com",
          "url": "http://example.com/currentServer",
        },
      ],
    },
  },
}
`;

exports[`diff parseData should parse data in correct order 1`] = `
Object {
  "candidateClient": Array [
    "candidateClient",
  ],
  "candidateServer": Array [
    "candidateServer",
  ],
  "currentClient": Array [
    "currentClient",
  ],
  "currentServer": Array [
    "currentServer",
  ],
}
`;

exports[`diff transformData should transform redirects 1`] = `
Object {
  "jsonLd": Object {},
  "metadata": Object {},
  "microdata": Object {},
  "redirects": Array [
    Object {
      "status": 301,
      "target": "http://example.com/test",
      "url": "http://example.com",
    },
  ],
}
`;
